hdat/store.py

```diff
@@ -17,21 +20,54 @@ def select(self, suite_id, case_id, result_id):
         else:
             return self.read_result(result_filename)
 
-    def select_all(self, *args):
+    def select_recent(self, i, *args):
         top_directory = os.path.join(self.root, *args)
-
-        result_filenames = []
-        for dirpath, _, filenames in os.walk(top_directory):
-            result_filenames.extend([os.path.join(dirpath, p) for p in filenames])
+        if not os.path.isdir(top_directory):
+            msg = "Selected case directory {} does not exist or is not a directory"
+            raise AbortError(msg.format(top_directory))
 
         results = []
-        for filename in result_filenames:
-            result = self.read_result(filename)
-            results.append(result)
+        ResultDesc = namedtuple('ResultDesc', ('id', 'ran_on'))
```

Comment:
Perhaps we should define this outside of the class?  I guess it is nice having it closer to the code that uses it, but its less efficient (but efficiency probably doesn't matter much here)...

Reply:
This is mostly for readability vs. a normal tuple here. Agree if this sorting case will show up elsewhere.